### ISSUE TITLE
Unit test `get_task_definition_image_url` from helpers/haws.py

### ISSUE THREAD

**Issue by samarth9008** [sorrentum/sorrentum#696]
We want to unit test `get_task_definition_image_url` from helpers/haws.py

The unit test file will helpers/test/test_haws.py

Make use [moto library](https://docs.getmoto.org/en/latest/) to mock aws services for the unit tests.

Write a test and run it locally inside docker container and make sure it passes. Create a PR once done

FYI @gpsaggese @sonaalKant

**Comment by 11swathi** [comment]
I mocked the aws service using moto and ran the unit test with pytest. 
Error 1:

```
FAILED test_haws.py::test_get_task_definition_image_url::test_get_task_definition_image_url_single_container - AssertionError: 
################################################################################
* Failed assertion *
File '/home/.aws/credentials' doesn't exist
```
Fix 1:
`mkdir -p ~/.aws && touch ~/.aws/credentials`

Error 2:

```
FAILED test_haws.py::Test_get_task_definition_image_url::test_get_task_definition_image_url - configparser.NoSectionError: No section: 'ck' 
```

Could you please help me with this? @Samarth

Code: 
```

from moto import mock_ecs
import boto3
import logging
import pytest
import os
import tempfile

import helpers.haws as haws
import helpers.hunit_test as hunitest

_LOG = logging.getLogger(__name__)

class Test_get_task_definition_image_url(hunitest.TestCase):

    @pytest.fixture(autouse=True, scope="class")
    def aws_credentials(self):
        """
        Mocked AWS Credentials for moto.
        """
        os.environ["MOCK_AWS_ACCESS_KEY_ID"] = "testing"
        os.environ["MOCK_AWS_SECRET_ACCESS_KEY"] = "testing"
        os.environ["MOCK_AWS_SECURITY_TOKEN"] = "testing"
        os.environ["MOCK_AWS_SESSION_TOKEN"] = "testing"
        os.environ["MOCK_AWS_DEFAULT_REGION"] = "us-east-1"

    @mock_ecs
    def test_get_task_definition_image_url(self):
        """
        Test that get_task_definition_image_url retrieves correct image URL.
        """
        aws_profile = "__mock__"
        task_definition_name = "test-task-definition"  

        # Creating a mock ECS task definition
        ecs = boto3.client("ecs", region_name="us-west-2")
        ecs.register_task_definition(
            family=task_definition_name,
            containerDefinitions=[
                {
                    "name": "test-container",
                    "image": "mock-image-url:latest"
                }
            ]
        )

        # Calling the function
        image_url = haws.get_task_definition_image_url(task_definition_name)

        # Assert
        self.assertEqual(image_url, "mock-image-url:latest")

    @mock_ecs
    def test_get_task_definition_image_url_nonexistent_definition(self):
        """
        Test that get_task_definition_image_url returns None for non-existent task definition.
        """
        aws_profile = "__mock__"
        task_definition_name = "non-existent-task-definition"  

        # Calling the function
        image_url = haws.get_task_definition_image_url(task_definition_name)

        # Assert
        self.assertIsNone(image_url)

    @mock_ecs
    def test_get_task_definition_image_url_no_image_url(self):
        """
        Test that get_task_definition_image_url returns None when there is no image URL.
        """
        aws_profile = "__mock__"
        task_definition_name = "no-image-url-task-definition"  

        # Creating a mock ECS task definition without image URL
        ecs = boto3.client("ecs", region_name="us-west-2")
        ecs.register_task_definition(
            family=task_definition_name,
            containerDefinitions=[
                {
                    "name": "test-container",
                }
            ]
        )

        # Calling the function
        image_url = haws.get_task_definition_image_url(task_definition_name)

        # Assert
        self.assertIsNone(image_url)

```

**Comment by sonaalKant** [comment]
Instead of copy pasting the code everytime, the suggested way is to create a draft PR and discuss on that PR. Also, while unit testing you might not have access to all the operations so we use https://docs.python.org/3/library/unittest.mock.html. 

We also have a documentation on this for reference https://github.com/sorrentum/sorrentum/blob/e841aaa34c71b200f5405e7257cc108a57086fd8/docs/coding/all.write_unit_tests.how_to_guide.md?plain=1#L758

**Comment by 11swathi** [comment]
> Instead of copy pasting the code everytime, the suggested way is to create a draft PR and discuss on that PR. 

Noted! I have created a PR - #711 for this task. Waiting for review. 
Thanks!

**Comment by samarth9008** [comment]
No need to ping us on issue. Instead add us as reviewers on PR itself and we will be notified.

**Comment by 11swathi** [comment]
Closed #696 as completed.
